tyr/servers/mongo/node.py

```diff
@@ -24,10 +25,17 @@ def __init__(self, group = None, server_type = None, instance_type = None,
                                         environment, ami, region, role,
                                         keypair, availability_zone,
                                         security_groups, block_devices,
-                                        chef_path)
+                                        chef_path, subnet_id)
 
     def configure(self):
 
+        self.security_groups = [
+            'management',
+            'chef-nodes',
+            '{envcl}'.format(envcl=self.envcl),
+            '{env}-mongo-management'.format(env=self.environment[0])
```

Comment:
Variables and properties like `self.envcl` and `self.environment` are being accessed at a time when they _may not_ be defined. If the user defines the `environment` argument there won't be an issue, but if they don't they'll get a `TypeError` because the `environment` variable isn't defined yet.